talismanrc/talismanrc.go

```diff
@@ -144,16 +170,68 @@ func IgnoreAdditionsByScope(additions []gitrepo.Addition, rcConfigIgnores Talism
 	}
 	return result
 }
+
+func (rcConfigIgnores *TalismanRCIgnore) AddFileIgnores(entriesToAdd []FileIgnoreConfig) {
+	if len(entriesToAdd) > 0 {
+		logr.Debugf("Adding entries: %v", entriesToAdd)
+		talismanRcIgnoreConfig := Get()
+		talismanRcIgnoreConfig.FileIgnoreConfig = combineFileIgnores(talismanRcIgnoreConfig.FileIgnoreConfig, entriesToAdd)
+		ignoreEntries, _ := yaml.Marshal(&talismanRcIgnoreConfig)
+		file, err := fs.OpenFile(currentRCFileName, os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			log.Printf("error opening %s: %s", currentRCFileName, err)
+		}
+		defer func() {
+			err := file.Close()
+			if err != nil {
+				log.Printf("error closing %s: %s", currentRCFileName, err)
+			}
+
+		}()
+		logr.Debugf("Writing talismanrc: %v", string(ignoreEntries))
+		_, err = file.WriteString(string(ignoreEntries))
+		if err != nil {
+			log.Printf("error writing to %s: %s", currentRCFileName, err)
+		}
+	}
+}
+
+func combineFileIgnores(exsiting, incoming []FileIgnoreConfig) []FileIgnoreConfig {
+	existingMap := make(map[string]FileIgnoreConfig)
+	for _, fIC := range exsiting {
+		existingMap[fIC.FileName] = fIC
+	}
+	for _, fIC := range incoming {
+		existingMap[fIC.FileName] = fIC
+	}
+	result := make([]FileIgnoreConfig, len(existingMap))
```

Comment:
Would appending the existing and incoming fileConfigs then sorting them by the fileName give the same result? 
allIgnores := append(existing, incoming)
return sort.SliceStable(allIgnores, func(i, j int) bool { return allIgnores[i].FileName < allIgnores[j].FileName })

Reply:
@tt-official : the map is used for removing duplicate entries. With the sorting approach we'd have to run an inner loop for each entry to remove duplicates, plus there is the complication of choosing which entry should be considered without context (unless the sorting function is stable, in which case the last entry can be used). Considering these things, I felt it was simpler to use a map and replace existing entries with the new entries from the incoming slice.